library/Rules/Json.php

```diff
@@ -15,11 +15,12 @@ class Json extends AbstractRule
 {
     public function validate($input)
     {
-        if (is_string($input)
-            && strtolower($input) == 'null') {
-            return true;
+        if (!is_string($input)) {
+            return false;
         }
 
-        return (null !== json_decode($input));
+        $decode = json_decode($input);
+
+        return is_object($decode) || is_array($decode);
```

Comment:
I would not do that. According to [json.org](http://www.json.org/), a JSON have three forms:
- An object is an unordered set of name/value pairs
- An array is an ordered collection of values
- A value can be a string in double quotes, or a number, or true or false or null, or an object or an array.

It doesn't need to be an array or an object, see this example:

```
php > var_dump(json_decode("Hello"));
NULL
php > var_dump(json_decode('"Hello"'));
string(5) "Hello"
```

See, that's because `"Hello"` is a valid Javascript notation.